### PR TITLE
Allow users to disable global popups.

### LINKED ISSUE #660: Allow disabling event popups in web UI
While I'm trying to perform a task via the web UI, Bodhi keeps showing pop-up notifications saying that someone has done something with some random update.  This is distracting and also completely pointless, since those events are not actionable or even relevant to me.  Can we have a way to turn this off?

popups

### PR DESCRIPTION
**Pull request by ralphbean**
Fixes #660.

People will still get popups about actions they're undertaking (like,
commenting on an update or error messages when they submit an incomplete form).

### CODE REVIEW

File: bodhi/models/models.py [hunk 1]
```diff
@@ -2007,6 +2008,9 @@ class User(Base):
     name = Column(Unicode(64), unique=True, nullable=False)
     email = Column(UnicodeText, unique=True)
 
+    # A preference
+    show_popups = Column(Boolean, default=True, server_default=text('TRUE'))
```
Comment (pypingou) [inline bodhi/models/models.py]: ``server_default=True`` doesn't work?

File: bodhi/views/generic.py [hunk 2]
```diff
@@ -193,6 +194,25 @@ def new_override(request):
...
+    from bodhi.models import User
+    userid = authenticated_userid(request)
+    if userid is None:
+        raise HTTPForbidden("You must be logged in.")
+    user = request.db.query(User).filter_by(name=unicode(userid)).first()
+    if user is None:
+        raise HTTPBadRequest("For some reason, user does not exist.")
+
+    # Toggle the value.
+    user.show_popups = not user.show_popups
```
Comment (pypingou) [inline bodhi/views/generic.py]: I guess we have autocommit on then?

Comment (ralphbean) [inline bodhi/views/generic.py]: We do, and it does.  :)

File: bodhi/models/models.py (hunk 1 above)
Comment (ralphbean) [inline bodhi/models/models.py]: Yeah, I tried ``server_default=True`` first and it balked.  I tried this a few times with upgrades and downgrades.

File: bodhi/views/generic.py [hunk 3]
```diff
@@ -193,6 +194,25 @@ def new_override(request):
...
 
 
+@view_config(route_name='popup_toggle', request_method='POST')
+def popup_toggle(request):
+    # Get the user
+    from bodhi.models import User
+    userid = authenticated_userid(request)
+    if userid is None:
+        raise HTTPForbidden("You must be logged in.")
+    user = request.db.query(User).filter_by(name=unicode(userid)).first()
```
Comment (lmacken) [inline bodhi/views/generic.py]: Can't we just do `request.user`?

Comment (ralphbean) [inline bodhi/views/generic.py]: Nah, ``request.user`` is a ``munch`` object now.  So, we can *get* the ``show_popups`` value from it, but we can't *set* it.

Comment (lmacken) [inline bodhi/views/generic.py]: Ah, gotcha. LGTM then! :+1:

**Comment by lmacken** [comment]
:fire: :dancers: :boom:

**Pull request merged**